src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -317,19 +318,20 @@ public void buttonAddCvrFileClicked() {
     String cvrFilePath = textFieldCvrFilePath.getText().trim();
     String cvrFirstVoteCol = textFieldCvrFirstVoteCol.getText().trim();
     String cvrFirstVoteRow = textFieldCvrFirstVoteRow.getText().trim();
+    String cvrProvider = textFieldCvrProvider.getText().trim();
     if (cvrFilePath.isEmpty()) {
       Logger.log(Level.WARNING, "CVR file path is required!");
-    } else if (cvrFirstVoteCol.isEmpty()) {
+    } else if (cvrFirstVoteCol.isEmpty() && !cvrProvider.equals("CDF")) {
```

Comment:
This seems like a really bizarre UX to me to have to count on the user to type "CDF" exactly in the provider box in order to skip needing to enter row and column fields, especially considering provider is an optional field. Why not instead just check if the file type is .json?

Reply:
OK, that makes sense here.